### PR TITLE
Add todo editing functionality

### PR DESCRIPTION
**Pull request by Tottoroo-Okarin**

### CODE REVIEW

File: client-react/src/features/todos/components/TodoActions/TodoActions.tsx [hunk 1]
```diff
@@ -1,25 +1,34 @@
...
-            </button>
-          )
-        }))
-      }
-    </div>
+    <>
+      {(editTodo) &&<Button onClick={() => openEditModal(todo)}>
+        Edit Todo
+      </Button>}
+      {(toggleTodo) && <Button onClick={() => toggleTodo(todo)}>
```
Comment (MursalovAltun) [inline client-react/src/features/todos/components/TodoActions/TodoActions.tsx:29]: Why do we render toggle action if there's a toggleTodo function? This always will be evaluated to true.

File: client-react/src/features/todos/components/TodoModal/TodoModal.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  boxShadow: 24,
+  p: 4,
+};
+
+export const TodoModal = ({isOpen}: TodoModalProps) => {
+  const todoBeingEdited = useAppSelector(TodoSelectors.todoBeingEdited)
+  const { register, handleSubmit, reset, formState: { errors } } = useForm<TodoInputs>();
+  const dispatch = useAppDispatch()
+
+  const clearFormState: TodoInputs = {
```
Comment (MursalovAltun) [inline client-react/src/features/todos/components/TodoModal/TodoModal.tsx:34]: This is a constant object that doesn't needs to be reinitialized each time our component is rerendered. Move this outside of the component.

File: client-react/src/features/todos/models/todoAction.model.ts [hunk 3]
```diff
@@ -1,4 +1,6 @@
+
```
Comment (MursalovAltun) [inline client-react/src/features/todos/models/todoAction.model.ts:1]: Keep naming convension the same accross the project. Use kebab-case

File: client-react/src/features/todos/models/todoInputs.model.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export interface TodoInputs {
```
Comment (MursalovAltun) [inline client-react/src/features/todos/models/todoInputs.model.ts:1]: Keep naming convension the same accross the project. Use kebab-case

File: client-react/src/features/todos/store/actions/todoModals.actions.ts [hunk 5]
```diff
@@ -2,12 +2,8 @@ import { createAction } from "@reduxjs/toolkit";
 import { Todo } from "../../models/todo.model";
```
Comment (MursalovAltun) [inline client-react/src/features/todos/store/actions/todoModals.actions.ts:2]: Keep naming convension the same accross the project. Use kebab-case

**Review by MursalovAltun — COMMENTED** [review]
See comments above

File: client-react/src/features/todos/components/TodoActions/TodoActions.tsx (hunk 1 above)
Comment (Tottoroo-Okarin) [reply client-react/src/features/todos/components/TodoActions/TodoActions.tsx:29]: ok zoomer

File: client-react/src/features/todos/enums/modalMode.enum.ts [hunk 6]
```diff
@@ -0,0 +1,4 @@
+export enum ModalMode {
```
[the remark this answers is not in the document]
Comment (Tottoroo-Okarin) [reply client-react/src/features/todos/enums/modalMode.enum.ts]: fixed

File: client-react/src/features/todos/components/TodoModal/TodoModal.tsx (hunk 2 above)
Comment (Tottoroo-Okarin) [reply client-react/src/features/todos/components/TodoModal/TodoModal.tsx]: fixed

File: client-react/src/features/todos/models/todoAction.model.ts (hunk 3 above)
Comment (Tottoroo-Okarin) [reply client-react/src/features/todos/models/todoAction.model.ts]: fixed

File: client-react/src/features/todos/store/actions/todoModals.actions.ts (hunk 5 above)
Comment (Tottoroo-Okarin) [reply client-react/src/features/todos/store/actions/todoModals.actions.ts]: fixed

File: client-react/src/features/todos/models/todoInputs.model.ts (hunk 4 above)
Comment (Tottoroo-Okarin) [reply client-react/src/features/todos/models/todoInputs.model.ts]: fixed

**Review by MursalovAltun — APPROVED** [review]

**Pull request merged**